### PR TITLE
Implement drop table migrations

### PR DESCRIPTION
**Pull request by 23Skidoo**
Plus add a basic test suite. [Case 2582](https://scrive.fogbugz.com/f/cases/2582).

### CODE REVIEW

**Comment by 23Skidoo** [comment]
Current status: @mariuszrak provided me with some feedback and requests for changes that I'm going to act on in the next few days.

**Comment by 23Skidoo** [comment]
Status update: delayed a bit.

**Comment by 23Skidoo** [comment]
This is done, BTW, just needs CR.

**Comment by mariuszrak** [comment]
1. I added a commit 0b83f28 with extra unit test. I believe it represents a real scenario that we will hit with, for example, CGI instance. Can we try to adjust branch so scenario described by this test is supported?
 
2. Can we change name of data MigrationType m  to MigrationAction and name of  mgrType to mgrAction? I always get confused with things that are names “type” that hold action inside.

3. If we are changing Migration structure - can we change mgrTable :: Table to mgrTable :: RawSQL (). I have a problem with migrations referring a table when I want to remove table definition, but keep migrations.
 
4. Whole Database.PostgreSQL.PQTypes.Checks module is quite heavy now.  It just very hard to follow up on code there - specially inside checkDBConsistency function. I don’t know how to solve it, but maybe you do. Maybe splitting it into smaller files would help?
 
5. Whole test setup is one big/long test that checks various things. So it’s hard to add new test - because whole setup is one-test-only fight now. Adding a second, even very simple test, would help a lot.

Please don't merge at the end with master - as I will want to adjust kontrakcja and do some migration testing on my machine.

**Comment by 23Skidoo** [comment]
Implemented the requested changes. The `Checks` module could use some more refactoring, and the test suite could be extended a little bit, though. I can look into this when I'll have time later this week; otherwise should be ready for re-review.

**Comment by mariuszrak** [comment]
This looks fine, but I'm doing more work to make this work with kontrakcja on other branch. I'll probably get back to this later this week

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# hpqtypes-extras-1.3.0.0 (2017-??-??)
```
Comment (jonathanjouty) [inline CHANGELOG.md]: @23Skidoo @mariuszrak might want to update this :)

Comment (mariuszrak) [inline CHANGELOG.md]: Done
